src/cc2olx/tools/README.rst

```diff
@@ -0,0 +1,61 @@
+
+The ``course_id`` argument is the ID of the course as it appears in Studio. For example, ``course-v1:edX+111222+111222``.
+
+The ``directory`` argument is a directory containing video files that will be uploaded to edX's video encoding pipeline.
+The currently supported formats for the videos are ``mp4`` and ``mov``, which are the files supported by the edX video encoding pipeline.
+Note that files of other types will be ignored by the tool.
+
+The ``csv`` argument is a file that is meant to establish a relationship between a particular video, its externally hosted URL, and the edX video ID that is generated  by the video encoding pipeline when the video is uploaded.
+
+In order to for this tool to run successfully, the following must be true of the CSV.
+
+* Each row in the file should describe a single video that appears in the directory supplied as a command line argument.
+* The CSV should contain a header row, labeling each column.
+* The first column should be labeled "Relative File Path". The values for this column should be the paths to the video files relative to the directory supplied as a command line argument. For example, if the directory is ``/Users/example/workspaces/videos`` and the video is located at ``/Users/example/workspaces/videos/unit_1/video_1.mp4``, the value of this cell should be ``unit_1/video_1.mp4``.
+* The CSV should contain the columns "External Video Link" and "Youtube ID". They can appear anywhere in the file as long as "Relative File Path" remains the first column.
+* Empty values should be represented as empty cells. Do not use "NONE", "N/A", etc.
+
+Output
+------
+
+The tool will generate a CSV. The name of the CSV will be the name of the original CSV supplied as a command line argument with "upload-results" appended to the end of the name, i.e. ``<original-filename>-upload-results.csv``.
+
+A new column, "Edx Id", which represents the edX Video ID for each video, will be appended to the end of the file.
+Otherwise, the file is identical to the file supplied as a command line argument.
+
+
+Logs
+----
+The tool will log to the console any errors that are encountered during execution.
+
+You can redirect the output to a file for later inspection.::
+
+    python src/cc2olx/tools/video_upload.py course-v1:edX+111222+111222 /Users/example/workspaces/videos/Users/example/workspaces/video-data.csv > logs.txt
+
+Use
+-----
+The video upload tool is a command-line Python 3 program.
+
+In order to make authenticated calls to Studio to generate signed URLs for uploading videos, you need to supply valid credentials.
+
+If you are an edX staff user, you can use the client credentials associated with an oAuth2 application registered in the desired environment. If you are not an edX staff user, you will have to go through the process for generating a client ID and client secret. You can use the `instructions <https://course-catalog-api-guide.readthedocs.io/en/latest/authentication/#getting-a-client-id-and-client-secret
```

Comment:
I'm not positive about this. Alan was handling authentication, so I need to look into this more to make sure these are the correct instructions.